### PR TITLE
Fix various sentry errors

### LINKED ISSUE #6983: OSError: [WinError 123] The filename, directory name, or volume label syntax is incorrect: '.D:'
Sentry Issue: [RASA-OPEN-SOURCE-3Y](https://sentry.io/organizations/rasahq/issues/1906835772/?referrer=github_integration)

```
OSError: [WinError 123] The filename, directory name, or volume label syntax is incorrect: '.D:'
(4 additional frame(s) were not displayed)
...
  File "rasa\__main__.py", line 116, in main
    cmdline_arguments.func(cmdline_arguments)
  File "rasa\cli\scaffold.py", line 208, in run
    _ask_create_path(path)
  File "rasa\cli\scaffold.py", line 155, in _ask_create_path
    os.makedirs(path)
```

### LINKED ISSUE #7046: AttributeError: 'Scope' object has no attribute 'set_user'
Sentry Issue: [RASA-OPEN-SOURCE-XP](https://sentry.io/organizations/rasahq/issues/1956318795/?referrer=github_integration)

```
AttributeError: 'Scope' object has no attribute 'set_user'
  File "rasa/__main__.py", line 115, in main
    rasa.telemetry.initialize_error_reporting()
  File "rasa/telemetry.py", line 232, in decorated
    return f(*args, **kwargs)
  File "rasa/telemetry.py", line 646, in initialize_error_reporting
    scope.set_user({"id": telemetry_id})
```

### LINKED ISSUE #7047: NotImplementedError: typ "['safe', 'rt']"not recognised (need to install plug-in?)
Sentry Issue: [RASA-OPEN-SOURCE-QR](https://sentry.io/organizations/rasahq/issues/1946798322/?referrer=github_integration)

```
NotImplementedError: typ "['safe', 'rt']"not recognised (need to install plug-in?)
(20 additional frame(s) were not displayed)
...
  File "rasa\shared\core\domain.py", line 131, in from_path
    domain = cls.from_file(path)
  File "rasa\shared\core\domain.py", line 144, in from_file
    return cls.from_yaml(rasa.shared.utils.io.read_file(path), path)
  File "rasa\shared\core\domain.py", line 150, in from_yaml
    yaml, rasa.shared.constants.DOMAIN_SCHEMA_FILE
  File "rasa\shared\utils\validation.py", line 140, in validate_yaml_schema
    yaml_file_content, reader_type=["safe", "rt"]
  File "rasa\shared\utils\io.py", line 247, in read_yaml
    yaml_parser = yaml.YAML(typ=reader_type)
```

### LINKED ISSUE #7048: TypeError: argument of type 'NoneType' is not iterable
Sentry Issue: [RASA-OPEN-SOURCE-FA](https://sentry.io/organizations/rasahq/issues/1925380037/?referrer=github_integration)

```
TypeError: argument of type 'NoneType' is not iterable
(12 additional frame(s) were not displayed)
...
  File "rasa\nlu\train.py", line 97, in train
    trainer = Trainer(nlu_config, component_builder)
  File "rasa\nlu\model.py", line 157, in __init__
    components.validate_requirements(cfg.component_names)
  File "rasa\nlu\components.py", line 54, in validate_requirements
    component_class = registry.get_component_class(component_name)
  File "rasa\nlu\registry.py", line 121, in get_component_class
    return rasa.shared.utils.common.class_from_module_path(component_name)
  File "rasa\shared\utils\common.py", line 17, in class_from_module_path
    if "." in module_path:
```

### LINKED ISSUE #7050: UnicodeDecodeError: 'utf-8' codec can't decode byte 0xa0 in position 576: invalid start byte
Sentry Issue: [RASA-OPEN-SOURCE-T7](https://sentry.io/organizations/rasahq/issues/1952329126/?referrer=github_integration)

```
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xa0 in position 576: invalid start byte
  File "rasa/shared/nlu/training_data/loading.py", line 134, in guess_format
    content = rasa.shared.utils.io.read_file(filename)
  File "rasa/shared/utils/io.py", line 118, in read_file
    return f.read()

UnicodeDecodeError: 'utf-8' codec can't decode byte 0xa0 in position 576: invalid start byte
(15 additional frame(s) were not displayed)
...
  File "rasa/shared/importers/multi_project.py", line 106, in _init_from_directory
    elif rasa.shared.data.is_nlu_file(full_path):
  File "rasa/shared/data.py", line 201, in is_nlu_file
    return nlu_loading.guess_format(file_path) != nlu_loading.UNK
  File "rasa/shared/nlu/training_data/loading.py", line 137, in guess_format
    if MarkdownReader.is_markdown_nlu_file(filename):
  File "rasa/shared/nlu/training_data/formats/markdown.py", line 177, in is_markdown_nlu_file
    content = rasa.shared.utils.io.read_file(filename)
  File "rasa/shared/utils/io.py", line 118, in read_file
    return f.read()
```

### LINKED ISSUE #7059: ValueError: NLU model predicted a wrong intent. Failed Story:
Sentry Issue: [RASA-OPEN-SOURCE-P0](https://sentry.io/organizations/rasahq/issues/1943109341/?referrer=github_integration)

```
ValueError: NLU model predicted a wrong intent. Failed Story: 

version: "2.0"
stories:
...

(6 additional frame(s) were not displayed)
...
  File "rasa/utils/common.py", line 308, in run_in_loop
    result = loop.run_until_complete(f)
  File "rasa/core/test.py", line 701, in test
    completed_trackers, agent, fail_on_prediction_errors, e2e
  File "rasa/core/test.py", line 605, in _collect_story_predictions
    tracker, agent, fail_on_prediction_errors, use_e2e
  File "rasa/core/test.py", line 552, in _predict_tracker_actions
    event, predicted, partial_tracker, fail_on_prediction_errors
  File "rasa/core/test.py", line 381, in _collect_user_uttered_predictions
    YAMLStoryWriter().dumps(partial_tracker.as_story().story_steps)
```

### PR DESCRIPTION
**Pull request by tmbo**
**Proposed changes**:
- fixes several sentry errors, one per commit, so probably easiest to review per commit 
- add telemetry context to sentries reported exceptions 85263f0

**Included fixes**:
- require minimal sentry version to avoid property issues. fixes #7046
- fixed user error in entering project directory. fixes #6983
- increased minimum ruamel version. fixes #7047
- raise invalid config error on components without a name. fixes #7048
- fixed unicode decode error throwing helpfull error for user. fixes #7050
- avoid raising valuerror on wrong predictions during test. fixes #7059

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by tmbo** [comment]
the only one changing user behaviour is c658aa8

File: changelog/7057.misc.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Fixed several sentry errors by introducing proper `RasaException`s.
```
Comment (federicotdn) [inline changelog/7057.misc.md:1]:
```suggestion
Fixed several Sentry errors by introducing proper `RasaException`s.
```

File: rasa/nlu/components.py [hunk 2]
```diff
@@ -51,6 +51,11 @@ def validate_requirements(component_names: List[Text]) -> None:
     # Validate that all required packages are installed
     failed_imports = {}
     for component_name in component_names:
+        if component_name is None:
+            raise InvalidConfigError(
```
Comment (federicotdn) [inline rasa/nlu/components.py:55]: We should add a `Raises:` section to this function's docstring.

File: rasa/telemetry.py [hunk 3]
```diff
@@ -645,6 +645,13 @@ def initialize_error_reporting() -> None:
         with configure_scope() as scope:
             scope.set_user({"id": telemetry_id})
 
+            default_context = _default_context_fields()
+            if "os" in default_context:
+                # os is a nested dict, hence we report it separately
+                scope.set_context("Operating System", default_context["os"])
+                del default_context["os"]
```
Comment (federicotdn) [inline rasa/telemetry.py:652]:
```suggestion
                scope.set_context("Operating System", default_context.pop("os"))
```

File: tests/nlu/test_components.py [hunk 4]
```diff
@@ -172,3 +172,17 @@ def test_can_handle_language_guard_clause(
     with pytest.raises(RasaException) as excinfo:
         SampleComponent.can_handle_language("random_string")
     assert expected_exec_msg in str(excinfo.value)
+
+
+async def test_validate_requirements_raises_exception_on_component_without_name(
+    tmp_path,
```
Comment (federicotdn) [inline tests/nlu/test_components.py:178]: Missing the type for `tmp_path` here.

**Review by federicotdn — APPROVED** [review]
Looks good! Only added small comments.

File: tests/nlu/test_components.py (hunk 4 above)
Comment (tmbo) [reply tests/nlu/test_components.py:178]: good point, added them to all tests in this file while I was at it 🙂

**Pull request merged**